logic/src/commonMain/kotlin/com/wire/kalium/logic/network/SessionManagerImpl.kt

```diff
@@ -60,4 +66,15 @@ class SessionManagerImpl(
     override suspend fun onClientRemoved() {
         sessionRepository.logout(userId, LogoutReason.REMOVED_CLIENT)
     }
+
+    override fun proxyCredentials(): ProxyCredentialsDTO? =
+        wrapStorageNullableRequest { proxyCredentialsStorage.fetch() }.nullableFold({
```

Comment:
We have the nullableFold and again in the next part we have the null check, what's the reason? I think I missed something here.